pkg/nsx/services/staticroute/staticroute_test.go

```diff
@@ -338,10 +338,11 @@ func TestStaticRouteService_Cleanup(t *testing.T) {
 		Path: &staticRoutePath2,
 	}
 
-	service.StaticRouteStore.Add(staticRoute1)
-	service.StaticRouteStore.Add(staticRoute2)
-
 	t.Run("Successful cleanup", func(t *testing.T) {
+		service.StaticRouteStore.Add(staticRoute1)
+		service.StaticRouteStore.Add(staticRoute2)
+		mockStaticRouteclient = mocks.NewMockStaticRoutesClient(mockController)
+		service.NSXClient.StaticRouteClient = mockStaticRouteclient
```

Comment:
Why this change will fix random failure? Could you explain this on PR description?

Reply:
reset the mockStaticRouteclient so the status will be reset